File: solutions/python/high-scores/1/high_scores.py
class HighScores:
    def __init__(self, scores):
        self.scores = scores

    def latest(self):
        return self.scores[-1] if len(self.scores) != 0 else None

    def personal_best(self):
        return max(self.scores)

    def personal_top_three(self):
        return sorted(self.scores, reverse = True)[:3]

File: solutions/java/high-scores/3/src/main/java/HighScores.java
import java.util.List;
import java.util.ArrayList;
import java.util.Comparator;

class HighScores {
    private List<Integer> scores;

    public HighScores(List<Integer> highScores) {
        this.scores = highScores;
    }

    List<Integer> scores() {
        return this.scores;
    }

    Integer latest() {
        int length = this.scores.size();
        return this.scores.get(length - 1);
    }

    Integer personalBest() {
        if (this.scores.isEmpty()){
            return null;
        }
        return personalTopThree().get(0);
    }

    List<Integer> personalTopThree() {
        return scores.stream()
                .sorted(Comparator.reverseOrder())
                .limit(3)
                .toList();
    }
}
